File: LocadoraVeiculos-FrontEnd/src/app/model/Reserva.ts
import { Cliente } from "./Cliente";
import { StatusReserva } from "./StatusReserva";
import { Veiculo } from "./Veiculo";

export class Reserva{
    id: number = 0;
    dtInicio: Date = new Date();
    dtFim: Date = new Date();
    valor: number = 0;
    status?: StatusReserva; 
    ClienteID: number = 0;
    cliente?: Cliente;
    veiculoID: number = 0;
    veiculo?: Veiculo;
}

File: LocadoraVeiculos/Models/Reserva.cs
using System.Text.Json.Serialization;

namespace LocadoraVeiculos.Models {
	public class Reserva {

		public int Id { get; set; }
		public DateTime dtInicio { get; set; }
		public DateTime dtFim { get; set; }
		public double valor { get; set; }
		[JsonConverter(typeof(JsonStringEnumConverter))]
		public StatusReserva? status { get; set; }
		public int ClienteID { get; set; }
		public Cliente? cliente { get; set; }
		public int VeiculoID { get; set; }
		public Veiculo? veiculo { get; set; }

		public Reserva() {
			status = StatusReserva.Andamento;
		}

	}
}
